File: parser/src/main/java/com/github/nsc/de/shake/parser/node/ImportNode.java
package com.github.nsc.de.shake.parser.node;

import com.github.nsc.de.shake.util.characterinput.position.PositionMap;

public class ImportNode extends Node {

    public static final String EVERYTHING = "*";

    private final String[] imported;

    public ImportNode(PositionMap map, String[] imported) {
        super(map);
        this.imported = imported;
    }

    public String[] getImport() {
        return imported;
    }
}

File: parser/src/main/kotlin/com/github/nsc/de/shake/parser/node/ImportNode.kt
package com.github.nsc.de.shake.parser.node

import com.github.nsc.de.shake.util.characterinput.position.PositionMap

class ImportNode(map: PositionMap, val import: Array<String>) : Node(map) {

    companion object {
        const val EVERYTHING = "*"
    }

    override fun toJson(): Map<String, *> =
        mapOf(
            "name" to "ImportNode",
            "import" to import
        )
}